lib/modifications.rb

```diff
@@ -29,74 +35,58 @@ def self.modify!(results)
         bundle.entry.first.resource.address.first.period.start = bundle.entry.first.resource.birthDate
       end
 
-      # Add discharge disposition to first encounter of each record
+      # Make sure there aren't stupid numbers of every resource type
+      missing_profiles = DataScript::Constraints::REQUIRED_PROFILES.dup
       results.each do |bundle|
-        encounter_entry = bundle.entry.find {|e| e.resource.resourceType == 'Encounter'}
-        encounter = encounter_entry.resource
-        encounter.hospitalization = FHIR::Encounter::Hospitalization.new
-        encounter.hospitalization.dischargeDisposition = create_codeable_concept('http://www.nubc.org/patient-discharge','01','Discharged to home care or self care (routine discharge)')
+        resource_counts = get_resource_counts(bundle).sort
+        # Move DocumentReferences to the front, so we delete them first (and don't have reference issues)
+        resource_counts.insert(0, resource_counts.delete(resource_counts.find { |k, _| k == 'DocumentReference' }))
```

Comment:
Exactly what is happening here and in line 45 is a bit opaque to me, but it seems odd to go
```
hash.find { |k, _| key == 'something' }
```
rather than just `hash['something']`.

Reply:
Technically `resource_counts` at this point is an array of arrays (thanks to `Hash#sort` in the line above), so `hash['something']` doesn't work in this context. I'll change the `find` variable names to better represent that fact (maybe something like `resource_name` and `_count`)